src/components/Main/Results/DeterministicLinePlot.tsx

```diff
@@ -248,103 +253,111 @@ export function DeterministicLinePlot({
 
   const yTickFormatter = (value: number) => formatNumberRounded(value)
 
-  return (
-    <div className="w-100 h-100" data-testid="DeterministicLinePlot">
-      <ReactResizeDetector handleWidth handleHeight>
-        {({ width }: { width?: number }) => {
-          if (!width) {
-            return <div className="w-100 h-100" />
-          }
-
-          const height = Math.max(500, width / ASPECT_RATIO)
-          const tooltipPosition = calculatePosition(height)
-
-          return (
-            <>
-              <h3>{t('Cases through time')}</h3>
-
-              <div ref={chartRef} />
-              <ComposedChart
-                onClick={() => scrollToRef(chartRef)}
-                width={width}
-                height={height}
-                data={plotData}
-                throttleDelay={75}
-                margin={{
-                  left: 15,
-                  right: 15,
-                  bottom: 15,
-                  top: 15,
-                }}
-              >
-                <CartesianGrid strokeDasharray="3 3" />
-
-                <XAxis
-                  dataKey="time"
-                  type="number"
-                  tickFormatter={xTickFormatter}
-                  domain={[tMin, tMax]}
-                  tickCount={7}
-                />
-
-                <YAxis scale={logScaleString} type="number" domain={[1, 'dataMax']} tickFormatter={yTickFormatter} />
-
-                <YAxis yAxisId="mitigationStrengthAxis" orientation={'right'} type="number" domain={[0, 1]} />
-
-                <Tooltip
-                  formatter={tooltipFormatter}
-                  labelFormatter={labelFormatter}
-                  position={tooltipPosition}
-                  content={ResponsiveTooltipContent}
-                />
-
-                <Legend
-                  verticalAlign="top"
-                  formatter={(v, e) => legendFormatter(enabledPlots, v, e)}
-                  onClick={(e) => {
-                    const plots = enabledPlots.slice(0)
-                    enabledPlots.includes(e.dataKey) ? plots.splice(plots.indexOf(e.dataKey), 1) : plots.push(e.dataKey)
-                    setEnabledPlots(plots)
+  function render() {
+    return (
```

Comment:
Here the original component's 'return' statement is wrapped into a function which will be debounced.